ReplayKitExample/ReplayKitExample/ViewController.swift

```diff
@@ -0,0 +1,398 @@
+    static let kRecordingAvailableInfo = "Ready to share the screen in a Broadcast (extension), or Conference (in-app)."
+    static let kRecordingNotAvailableInfo = "ReplayKit is not available at the moment. Another app might be recording, or AirPlay may be in use."
+
+    // An application has a much higher memory limit than an extension. You may choose to deliver full sized buffers instead.
+    static let kDownscaleBuffers = false
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        broadcastButton.setTitle(ViewController.kStartBroadcastButtonTitle, for: .normal)
+        conferenceButton?.setTitle(ViewController.kStartConferenceButtonTitle, for: .normal)
+        broadcastButton.layer.cornerRadius = 4
+        conferenceButton?.layer.cornerRadius = 4
+
+        // The setter fires an availability changed event, but we check rather than rely on this implementation detail.
+        RPScreenRecorder.shared().delegate = self
+        checkRecordingAvailability()
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIScreenCapturedDidChange, object: UIScreen.main, queue: OperationQueue.main) { (notification) in
+            if self.broadcastPickerView != nil && self.screenTrack == nil {
+                let title = UIScreen.main.isCaptured ? ViewController.kInProgressBroadcastButtonTitle : ViewController.kStartBroadcastButtonTitle
+                self.broadcastButton.setTitle(title, for: .normal)
+                self.conferenceButton?.isEnabled = !UIScreen.main.isCaptured
+            }
+        }
+
+        // Use RPSystemBroadcastPickerView when available (iOS 12+ devices).
+        // TODO: Use #if targetEnvironment(simulator) after upgrading the examples to Swift 4.2.
+        #if arch(arm64)
+        if #available(iOS 12.0, *) {
+            // Swap the button for an RPSystemBroadcastPickerView.
+            let pickerView = RPSystemBroadcastPickerView(frame: CGRect(x: 0,
+                                                                       y: 0,
+                                                                   width: view.bounds.width,
+                                                                  height: 80))
+            pickerView.preferredExtension = "com.twilio.ReplayKitExample.BroadcastVideoExtension"
+            view.addSubview(pickerView)
+
+            self.broadcastPickerView = pickerView
+            broadcastButton.isEnabled = false
+        }
+        #endif
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+
+        // Our picker will be the same size as the hidden button it replaces.
+        if let picker = self.broadcastPickerView {
+            picker.frame = self.broadcastButton.frame.offsetBy(dx: 0, dy: -10)
+            self.broadcastButton.titleEdgeInsets = UIEdgeInsets(top: 34, left: 0, bottom: 0, right: 0)
+        }
```

Comment:
This code is problematic, I see the picker go crazy when rotating the screen.

Reply:
will setting of the picker with the button help?